warehouse/bigquery_test.go

```diff
@@ -0,0 +1,132 @@
+package warehouse
+
+import (
+	"testing"
+
+	"cloud.google.com/go/bigquery"
+)
+
+var _ Warehouse = &BigQuery{}
+
+func TestGetMissingFields(t *testing.T) {
+	wh := &BigQuery{}
+
+	requiredField1 := bigquery.FieldSchema {
+		Name: "RequiredField1",
+	}
+	requiredField1Lowercase := bigquery.FieldSchema {
+		Name: "requiredfield1",
+	}
+	requiredField2 := bigquery.FieldSchema {
+		Name: "RequiredField2",
+	}
+	dummyField := bigquery.FieldSchema {
+		Name: "DummyField",
+	}
+
+	hauserSchema := bigquery.Schema([]*bigquery.FieldSchema {
+		&requiredField1,
+		&requiredField2,
+	})
+
+	var testCases = []struct {
+		hauserSchema	bigquery.Schema
+		tableSchema		bigquery.Schema
```

Comment:
nit: whitespace inconsistency, may be worth it to run `goimports` on this file. 

Last year when I first started doing go, I tried to hook this into my IDE, but it would get into weird reload from disk loops. What was then Gogland is now GoLand, and might be a bit better. I haven't tried it again, largely because of the tools already available in our other repos.

Reply:
For this, you want `gofmt -w -s`. BTW, I would highly recommend adding a CI build to this repo. Take a look at the `Makefile` and `.travis.yml` files in some of our other repos, like this one:
https://github.com/fullstorydev/grpcurl
(Obviously that should be separate PR -- maybe multiple PRs if you also need to change a decent bit of code to actually make all of the static checks pass.)